sentinel/src/lib/feed.js

```diff
@@ -38,6 +63,94 @@ const registerFeed = seq => {
     });
 };
 
+const deleteReadDocs = change => {
+  // we don't know if the deleted doc was a report or a message so
+  // attempt to delete both
+  const possibleReadDocIds = ['report', 'message'].map(
+    type => `read:${type}:${change.id}`
+  );
+
+  return db.allDbs().then(dbs => {
+    const userDbs = dbs.filter(dbName => dbName.indexOf(`${db.medicDbName}-user-`) === 0);
+    return userDbs.reduce((p, userDb) => {
+      return p.then(() => {
+        const metaDb = db.get(userDb);
+        return metaDb
+          .allDocs({ keys: possibleReadDocIds })
+          .then(results => {
+            const row = results.rows.find(row => !row.error);
+            if (!row) {
+              return;
+            }
+
+            return metaDb.remove(row.id, row.value.rev).catch(err => {
+              // ignore 404s or 409s - the doc was probably deleted client side already
+              if (err && err.status !== 404 && err.status !== 409) {
+                throw err;
+              }
+            });
+          })
+          .then(() => {
+            db.close(metaDb);
+          })
+          .catch(err => {
+            db.close(metaDb);
+            throw err;
+          });
+      });
+    }, Promise.resolve());
+
+  });
+};
+
+
+const changeQueue = async.queue((change, callback) => {
+  if (lastSeq && change.seq === lastSeq) {
+    lastSeq = undefined;
+    lastInQueue = true;
+  }
+
+  if (!change) {
+    return callback();
+  }
+  logger.debug(`change event on doc ${change.id} seq ${change.seq}`);
+  if (processed > 0 && processed % PROGRESS_REPORT_INTERVAL === 0) {
+    logger.info(
+      `transitions: ${processed} items processed (since sentinel started)`
+    );
+  }
+  if (change.deleted) {
+    // don't run transitions on deleted docs, but do clean up
+    return Promise
+      .all([
+        transitionsLib.infodoc.delete(change),
+        deleteReadDocs(change)
+      ])
+      .catch(err => {
+        logger.error('Error cleaning up deleted doc: %o', err);
+      })
+      .then(() => tombstoneUtils.processChange(Promise, db.medic, change, logger))
+      .then(() => updateMetadata(change, callback))
+      .catch(callback);
+  }
+
+  transitionsLib.processChange(change, err => {
+    if (err) {
+      return callback(err);
+    }
+
+    updateMetadata(change, callback);
+  });
+});
+
+changeQueue.drain(() => {
+  if (lastInQueue) {
+    lastInQueue = false;
+    logger.info(`transitions: queue drained, we restart the listener`);
+    listen();
+  }
+});
+
 const listen = () => {
   if (!init) {
     init = getProcessedSeq().then(seq => registerFeed(seq));
```

Comment:
You're not ever actually using `lastSeq`, just writing to it. This is where you should be using `lastSeq` if it's set, otherwise pulling the processed seq out of the metadata document